app/src/main/java/com/cornellappdev/android/eatery/data/repositories/UserPreferencesRepository.kt

```diff
@@ -64,6 +69,19 @@ class UserPreferencesRepository @Inject constructor(
         }
     }
 
+    fun toggleFavoriteMenuItem(menuItem: String) {
+        CoroutineScope(Dispatchers.IO).launch {
+            userPreferencesStore.updateData { currentPreferences ->
+                val isFavorite = currentPreferences.itemFavoritesMap[menuItem] == true
+                if (!isFavorite) {
+                    currentPreferences.toBuilder().putItemFavorites(menuItem, true).build()
+                } else {
+                    currentPreferences.toBuilder().removeItemFavorites(menuItem).build()
+                }
+            }
+        }
+    }
```

Comment:
Very minor nitpick, but let's make this a `suspend fun` without any Coroutine in there. I know when I was implementing some stuff in Eatery back in the day, I DIDNT do this, but I think it's very slightly better.

Suppose the VM wanted to wait until this line of code strictly succeeded/terminated before doing something. If this is a `suspend fun`, it can choose to block some line of code behind the execution here. But if it doesn't care, it can also just launch `viewModelScope` and then have that line of code outside on the coroutine block.

Again, super minor thing, but I think it's slightly better. And then of course you can just call this function by launching a `viewModelScope`.

What do u think?

Reply:
Yeah that's definitely my bad. Sometimes I just go with the status quo without thinking too deeply about it, but I should try to think more critically about stuff like this. I'll fix this up now.